C++ API parity: ELU

pbelevich opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27060 C++ API parity: LogSigmoid
* #27059 C++ API parity: LeakyReLU
* #27038 C++ API parity: Hardtanh
* #27035 C++ API parity: Hardshrink
* **#27028 C++ API parity: ELU**

Differential Revision: [D17682406](https://our.internmc.facebook.com/intern/diff/D17682406)

pbelevich committed:
C++ API parity: ELU

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: ELU"




[ghstack-poisoned]

yf225 commented on torch/csrc/api/include/torch/nn/functional/activation.h:
```diff
@@ -0,0 +1,20 @@
+
+inline Tensor elu(const Tensor& input, const ELUOptions& options) {
+  if (options.inplace()) {
+    return torch::elu_(const_cast<Tensor&>(input), options.alpha());
```
hmm I think we can consider changing `input`'s type from `const Tensor&` to `Tensor&`, since `elu` can be an in-place operation. After that change, we won't need `const_cast` here.

yf225 commented on torch/csrc/api/src/nn/modules/activation.cpp:
```diff
@@ -0,0 +1,24 @@
+void ELUImpl::pretty_print(std::ostream& stream) const {
+  stream << std::boolalpha
+         << "torch::nn::ELU(alpha=" << options.alpha()
+         << ", inplace=" << options.inplace() << ")";
```
In the Python version, it seems that whether it prints `inplace` is conditional:
```python
    def extra_repr(self):
        inplace_str = ', inplace=True' if self.inplace else ''
        return 'alpha={}{}'.format(self.alpha, inplace_str)
```
and I think we should do the same here.

yf225 commented on torch/csrc/api/include/torch/nn/options/activation.h:
```diff
@@ -0,0 +1,20 @@
+  // The alpha value for the ELU formulation. Default: 1.0
+  TORCH_ARG(double, alpha) = 1.0;
+
+  // can optionally do the operation in-place. Default: False
```
nit: need `///` instead of `//` (similarly for other PRs)

pbelevich committed:
Update on "C++ API parity: ELU"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: ELU"




[ghstack-poisoned]

yf225 reviewed (approved):
Thanks @pbelevich !

pbelevich committed:
Update on "C++ API parity: ELU"


Differential Revision: [D17682406](https://our.internmc.facebook.com/intern/diff/D17682406)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: ELU"


Differential Revision: [D17682406](https://our.internmc.facebook.com/intern/diff/D17682406)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: ELU"


Differential Revision: [D17682406](https://our.internmc.facebook.com/intern/diff/D17682406)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: ELU"


Differential Revision: [D17682406](https://our.internmc.facebook.com/intern/diff/D17682406)

[ghstack-poisoned]
